action/protocol/context.go

```diff
@@ -307,6 +311,13 @@ func WithFeatureCtx(ctx context.Context) context.Context {
 	)
 }
 
+func (fCtx *FeatureCtx) Tolerate(err error) bool {
+	if fCtx.TolerateEmptyCandidateName && errors.Cause(err) == action.ErrInvalidCanName {
+		return true
+	}
+	return false
+}
```

Comment:
error type shouldn't not be included in a function which doesn't name it's functionality. https://github.com/iotexproject/iotex-core/blob/5d5e5afc1e9afaddcdfaf24bb47acc00f5aed126/action/protocol/generic_validator.go#L61 is a better place to handle the check

Reply:
do you mean the func name is not clear? or should not check error type in this func?